### PR TITLE
py38: batch dependency upgrades

### PR DESCRIPTION
**Pull request by joshuarli**
These should be the remaining dependencies that need upgrades and most likely no code changes on our part.

Danny'll be testing python3-saml upgrade still works.

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -22,12 +22,12 @@ google-cloud-pubsub==2.2.0
...
 jsonschema==3.2.0
 lxml==4.6.3
-maxminddb==1.5.4
+maxminddb==2.0.3
 mistune==0.8.4
 mmh3==3.0.0
 parsimonious==0.8.0
 petname==2.6
-phonenumberslite==8.11.3
+phonenumberslite==8.12.0
```
Comment (joshuarli) [inline requirements-base.txt:30]: Just metadata updates, good phone numbers.

File: requirements-base.txt [hunk 2]
```diff
@@ -52,12 +52,12 @@ sentry-relay==0.8.5
 sentry-sdk>=1.0.0,<1.2.0
 snuba-sdk>=0.0.9,<1.0.0
 simplejson==3.17.2
-statsd==3.1
+statsd==3.3
```
Comment (joshuarli) [inline requirements-base.txt:55]: We don't use this in production, but we leave it as an option.

File: requirements-base.txt [hunk 3]
```diff
@@ -52,12 +52,12 @@ sentry-relay==0.8.5
...
 structlog==17.1.0
 symbolic==8.0.3
 toronado==0.1.0
 ua-parser==0.10.0
-unidiff==0.5.5
+unidiff==0.6.0
```
Comment (joshuarli) [inline requirements-base.txt:60]: Very trivial usage which changes don't really affect.

File: requirements-base.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ celery==4.4.7
 click==7.1.2
 confluent-kafka==1.5.0; python_version <= '3.8'
 confluent-kafka==1.6.0; python_version == '3.9'
-croniter==0.3.34
+croniter==0.3.37
```
Comment (joshuarli) [inline requirements-base.txt:8]: Trivial upstream changes.

File: requirements-base.txt [hunk 5]
```diff
@@ -77,8 +77,7 @@ grpcio==1.35.0
 hiredis==0.3.1
 
 # not directly used, but pinned for at least semaphore/symbolic
-cffi==1.12.2; python_version <= '3.8'
-cffi==1.14.4; python_version == '3.9'
+cffi==1.14.5
```
Comment (joshuarli) [inline requirements-base.txt:80]: Contains some good fixes for cpython 3.7.x, 3.8.x.

File: requirements-dev.txt [hunk 6]
```diff
@@ -10,4 +10,3 @@ pytest-django==3.10.0
 pytest-sentry==0.1.6
 pytest-rerunfailures==9.1.1
 responses==0.10.12
-sqlparse==0.2.4
```
Comment (joshuarli) [inline requirements-dev.txt:13]: Duplicated in requirements-base.

File: requirements-base.txt [hunk 7]
```diff
@@ -22,12 +22,12 @@ google-cloud-pubsub==2.2.0
 google-cloud-storage==1.35.0
 jsonschema==3.2.0
 lxml==4.6.3
-maxminddb==1.5.4
+maxminddb==2.0.3
```
Comment (joshuarli) [inline requirements-base.txt:25]: The [changes](https://github.com/maxmind/MaxMind-DB-Reader-python/blob/main/HISTORY.rst) are pretty trivial.

File: requirements-base.txt [hunk 8]
```diff
@@ -37,7 +37,7 @@ PyJWT==1.5.3
 python-dateutil==2.8.1
 python-memcached==1.59
 python-u2flib-server==5.0.0
-python3-saml==1.4.1
+python3-saml==1.10.1
```
Comment (joshuarli) [inline requirements-base.txt:40]: Leaving it to Danny to manually test that SAML2 still works.

**Review by armenzg — APPROVED** [review]

**Pull request merged**